templates/terraform/custom_flatten/id_from_name.erb

```diff
@@ -0,0 +1,18 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2017 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData) interface{} {
```

Comment:
Does this leave v unused?  Does that compile?

Reply:
Named return values that are unused are not compilation errors: https://play.golang.org/p/c-sftYU07rB

Reply:
Oops, that's an argument, not a named return value. Same deal: https://play.golang.org/p/zSxJpuwtOTj

Reply:
apparently I forgot to hit "add comment", but here is what was sitting in a text field for the last 2 hours:

go is fine with parameters that aren't used, even though it hates that situation with local variables. Several flatten functions don't use d, for instance https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_disk.go#L844-L846